app/helpers/format-currency.js

```diff
@@ -15,20 +24,16 @@ const COMMAS_EVERY_THREE_DIGITS = /(\d)(?=(\d{3})+(?!\d))/g;
  * - dot as decimal separator
  * - fixed 2-decimal notation
  *
- * @param  {Array[String]} params An array of paramaters provided to the helper in the template
- * @return {String}        Amount formated as $20,000.99
+ * In order to not display zero decimals, provide a hash, consisting of
+ * `{ trimZero: true }` as the second parameter.
+ *
+ * @param  {String}  dollarsAsString   The value to be formatted as currency
+ * @param  {Boolean} options.trimZero  Indicates if zero decimals should be removed
+ * @return {String}                    The value, formatted as currency
  */
-export function formatCurrency(params) {
-  let [dollarsAmountAsString] = params;
-  let dollarsAmount = parseFloat(dollarsAmountAsString);
-
-  if (dollarsAmount) {
-    return `$${dollarsAmount.toFixed(2).replace(COMMAS_EVERY_THREE_DIGITS, '$1,')}`;
-  } else if (isNaN(dollarsAmount)) {
-    return '';
-  } else {
-    return dollarsAmount;
-  }
+export function formatCurrency([dollarsAsString, { trimZero = false } = {}]) {
```

Comment:
You wouldn't use the hash helper when using `namedArgs`. It would be `{{format-currency amount trimZero=true}}`. Ember will put all named arguments into a hash for us and pass that as the second argument for us